Makefile

```diff
@@ -1,52 +1,191 @@
+FAIL	= (echo ${TIME} ${RED}[FAIL]${CNone} && false)
+
+# ====================================================================================
+# Targets
+
+help: ## to get help
+	@echo "Usage:"
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) |\
+	awk 'BEGIN {FS = ":.*?## "}; {printf "make \033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-.PHONY: build
-build:
-	mkdir -p dist
-	env GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o dist/rtcd -ldflags '$(LDFLAGS)' -mod=readonly -trimpath ./cmd/rtcd
+.PHONY: lint
+lint: go-lint go-mod-check docker-lint  ## to lint all
+
+.PHONY: test
+test: go-test  ## to test all
 
 .PHONY: docker-build
-docker-build:
-	@if ! [ -x "$$(command -v docker)" ]; then \
-		echo "Docker is not installed. Please see https://docs.docker.com/get-docker for installation instructions."; \
-		exit 127; \
-	fi; \
+docker-build: go-build ## to build the docker image
+	@$(INFO) Performing Docker build...
+	@docker build -f ${CONFIG_DOCKERFILE} . \
+	-t ${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@$(OK) Performing Docker build ${APP_NAME}:${APP_VERSION}
+
+.PHONY: docker-push
+docker-push: ## to push the docker image
+	@$(INFO) Pushing to registry...
+	@docker tag ${APP_NAME}:${APP_VERSION} $(CONFIG_IMAGE_REGISTRY)/${CONFIG_DOCKER_REPO}/${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@docker push $(CONFIG_IMAGE_REGISTRY)/${CONFIG_DOCKER_REPO}/${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@$(OK) Pushing to registry $(CONFIG_IMAGE_REGISTRY)/${CONFIG_DOCKER_REPO}/${APP_NAME}:${APP_VERSION}
+
+.PHONY: docker-sbom
+docker-sbom: ## to print a sbom report
+	@$(INFO) Performing Docker sbom report...
+	@docker sbom ${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@$(OK) Performing Docker sbom report
+
+.PHONY: docker-scan
+docker-scan: ## to print a vulnerability report
+	@$(INFO) Performing Docker scan report...
+	@docker scan ${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@$(OK) Performing Docker scan report
+
+.PHONY: docker-lint
+docker-lint: ## to lint the Dockerfile
+	@$(INFO) Dockerfile linting...
+	@docker run ${CONFIG_DOCKEROPTS} \
+	${IMAGE_DOCKERLINT} \
+	< ${CONFIG_DOCKERFILE} || ${FAIL}
+	@$(OK) Dockerfile linting
+
+go-build: $(GO_BUILD_PLATFORMS_ARTIFACTS) ## to build binaries
+
+.PHONY: go-build
+go-build/%:
+	@$(INFO) go build $*...
+	@target="$*"; \
+	command="$${target%%-*}"; \
+	platform_ext="$${target#*-}"; \
+	platform="$${platform_ext%.*}"; \
+	export GOOS="$${platform%%-*}"; \
+	export GOARCH="$${platform#*-}"; \
+	echo export GOOS=$${GOOS}; \
+	echo export GOARCH=$${GOARCH}; \
+	CGO_ENABLED=0 \
+	$(GO) build ${GO_BUILD_OPTS} \
+	-ldflags '${GO_LDFLAGS}' \
+	-o ${CONFIG_BIN_DIR}/$* \
+	${CONFIG_APP_CODE} || ${FAIL}
+	@$(OK) go build $*
+
+.PHONY: go-run
+go-run: ## to run locally for development
+	@$(INFO) running locally...
+	@$(GO) run ${GO_BUILD_OPTS} ${CONFIG_APP_CODE} || ${FAIL}
+	@$(OK) running locally
+
+.PHONY: go-test
+go-test: ## to run tests
+	@$(INFO) testing...
+	@$(GO) test ${GO_TEST_OPTS} ./... || ${FAIL}
+	@$(OK) testing
```

Comment:
How do you feel about printing out the command? Personally I am used to see which flags are passed.

Reply:
From my point of view, I would like to see output only if a target fails to execute. 
Maybe we can check and see if we can add more verbosity options in the makefile, in a next iteration. Would that work for you?

Reply:
Rebased and pushed changes